app/js/streaming/protection/ProtectionController.js

```diff
@@ -644,8 +644,8 @@ MediaPlayer.dependencies.ProtectionController = function() {
 
             this.keySystem = undefined;
 
+            this.protectionModel.teardown();
             this.setMediaElement(null).then(function() {
-                self.protectionModel.teardown();
```

Comment:
This will introduce some regression.
It is important to release video element before ProtectionModel. See commit 866b42580b33731515d760c9c63034b09d58db84